saleor/product/utils/variants.py

```diff
@@ -64,7 +64,7 @@ def fetch_variants_for_promotion_rules(
     for rule in rules.iterator():
         variants = get_variants_for_catalogue_predicate(
             rule.catalogue_predicate,
-            database_connection_name=settings.DATABASE_CONNECTION_REPLICA_NAME,
+            database_connection_name=database_connection_name,
```

Comment:
This will cause the celery task like, `set_promotion_rule_variants_task`, will start using `DEFAULT` database. IMHO, you should change the default `database_connection_name` to `REPLICA`

Reply:
Changed back